File: FeedBuilder/FileInfoEx.vb
Imports System.IO

Public Class FileInfoEx
	Private myFileInfo As FileInfo
	Private myFileVersion, myHash As String

	Public ReadOnly Property FileInfo As FileInfo
		Get
			Return myFileInfo
		End Get
	End Property

	Public ReadOnly Property FileVersion As String
		Get
			Return myFileVersion
		End Get
	End Property

	Public ReadOnly Property Hash As String
		Get
			Return myHash
		End Get
	End Property

	Public Sub New(fileName As String)
		myFileInfo = New FileInfo(fileName)
		myFileVersion = FileVersionInfo.GetVersionInfo(fileName).FileVersion
		myHash = NAppUpdate.Framework.Utils.FileChecksum.GetSHA256Checksum(fileName)
	End Sub
End Class

File: FeedBuilder/FileInfoEx.cs
using System.Diagnostics;
using System.IO;
using NAppUpdate.Framework.Utils;

namespace FeedBuilder
{
	public class FileInfoEx
	{
		private readonly FileInfo myFileInfo;
		private readonly string myFileVersion;
		private readonly string myHash;

		public FileInfo FileInfo
		{
			get { return myFileInfo; }
		}

		public string FileVersion
		{
			get { return myFileVersion; }
		}

		public string Hash
		{
			get { return myHash; }
		}

        public string RelativeName { get; private set; }

		public FileInfoEx(string fileName, int rootDirLength)
		{
			myFileInfo = new FileInfo(fileName);
            var verInfo = FileVersionInfo.GetVersionInfo(fileName);
            if (myFileVersion != null)
                myFileVersion = new System.Version(verInfo.FileMajorPart, verInfo.FileMinorPart, verInfo.FileBuildPart, verInfo.FilePrivatePart).ToString();
			myHash = NAppUpdate.Framework.Utils.FileChecksum.GetSHA256Checksum(fileName);
            RelativeName = fileName.Substring(rootDirLength + 1);
		}
	}
}
